catalog_integrations_api_tools/sftp.py

```diff
@@ -30,14 +53,38 @@ def get_all_files(
         Returns:
             str: Name of directory the files were downloaded to.
         """
-        pass
+        products_pattern_compiled = re.compile(
+            rf"{file_prefix}_products_\d+\.csv(\.gz)?", flags=re.I
+        )
+        stockprices_pattern_compiled = re.compile(
+            rf"{file_prefix}_stock-prices_\d+\.csv(\.gz)?", flags=re.I
+        )
+
+        kwargs = self._get_sftp_credentials()
+        with pysftp.Connection(**kwargs) as sftp:
+            with sftp.cd(remote_directory):
+                for filename in sftp.listdir():
+                    if re.match(products_pattern_compiled, filename) or re.match(
+                        stockprices_pattern_compiled, filename
+                    ):
+                        self.all_files.append(os.path.join(remote_directory, filename))
+                        sftp.get(filename, os.path.join(local_directory, filename))
+                        logger.info(f"File {filename} downloaded successfully.")
+        return local_directory
 
     def move_files(self, remote_directory: str = "/data/cornershop/processed") -> None:
         """
-        Move processed files to processed directory on SFTP server.
+        Move processed files to /data/cornershop/processed directory on SFTP server.
         This only moves the files that were the downloaded via the get_all_files method.
 
         Args:
             remote_directory (str, optional): Remote directory to move the files to. Defaults to "/data/cornershop/processed".
         """
-        pass
+        kwargs = self._get_sftp_credentials()
+        with pysftp.Connection(**kwargs) as sftp:
+            for filepath in self.all_files:
+                _, filename = os.path.split(filepath)
+                if sftp.isdir(remote_directory) is False:
+                    sftp.mkdir(remote_directory)
```

Comment:
I am not sure if we should assume all parent directories already exist.
Something to think about...
```suggestion
                    sftp.makedirs(remote_directory)
```

Reply:
I don't understand the assumption. Do you mean "/data/cornershop" does not exist?

Reply:
If the integrator wants to save move their files to `data/cornershop/files/processed` but the `data/cornershop/files` does not exist yet, it would need to be created before `data/cornershop/files/processed` this directory.
`sftp.mkdir` does not create it whereas `sftp.makedirs` does.

Reply:
I see your point. I'll make that change.